AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_LightAdded.py

```diff
@@ -212,33 +211,34 @@ def AtomEditorComponents_Light_AddedToEntity():
             # 5. Set light type.
             light_component.set_component_property_value(
                 AtomComponentProperties.light('Light type'), LIGHT_TYPES[light_type])
+            general.idle_wait_frames(1)
+            current_light_type = light_component.get_component_property_value(
+                              AtomComponentProperties.light('Light type'))
             test_light_type = (
                 f"Set light type: {light_type.upper()}",
                 f"P0: Light component failed to set {light_type.upper()} type")
-            general.idle_wait_frames(1)
             Report.result(test_light_type,
-                          light_component.get_component_property_value(
-                              AtomComponentProperties.light('Light type')) == LIGHT_TYPES[light_type])
+                          current_light_type == LIGHT_TYPES[light_type])
 
             # 6. Check for Shape component.
-            if light_type in ('sphere', 'spot_disk', 'capsule', 'quad', 'polygon'):
+            if current_light_type in (LIGHT_TYPES['sphere'], LIGHT_TYPES['spot_disk'], LIGHT_TYPES['capsule'],
```

Comment:
if LIGHT_TYPE[light_type] in
don't use the value from the component under test to branch your tests since the component may be giving an incorrect response.

Reply:
Updated